### ISSUE TITLE
How to specify Nextcloud version for upgrades

### ISSUE THREAD

**Issue by Phillomath** [nextcloud/docker#1042]
I have somehow ended up skipping past a major version of Nextcloud when executing a:
`docker-compose pull`
`docker-compose up -d`

And have gone straight from 16.0.1 to 18.0.2.  This has resulted in the error on the web interface:

`Internal Server Error`
`The server encountered an internal error and was unable to complete your request.
Please contact the server administrator if this error reappears multiple times, please include the technical details below in your report.
More details can be found in the server log.`

I have seen other people have resolved this by doing an occ upgrade from the command line interface, but I am trying to avoid that as it is not supported to skip a major version when upgrading the database.

It is my hope that my database has not yet been upgraded and that I can pull version 17, upgrade and then pull version 18 and upgrade again.

In my docker-compose.yml file the version for nextcloud is specified as  "nextcloud:apache", I have tried changing that to "nextcloud:17.0:apache" and "nextcloud:apache:17.0" but  get `ERROR: no such image: nextcloud:apache:17.0: invalid reference format`

How do I follow the recommended path for upgrading, or have I already borked my install and an occ upgrade is my best bet?

Cheers,

Phillomath

**Comment by kungknut** [comment]
Hi!

Have a look at the Supported tags section at docker hub, https://hub.docker.com/_/nextcloud.

I think you may be looking for this; `nextcloud:17.0-apache`

**Comment by wandhydrant** [comment]
I put myself in a similar situation (I always used "stable" when it would be better to ensure step by step updates). This howto got me out of it:
https://help.nextcloud.com/t/updates-between-multiple-major-versions-are-unsupported/7094

Basically, from the html/data volume, I only left "data" and "config" and then re-created the container.

ALWAYS backup both database and files before upgrading!

**Comment by Phillomath** [comment]
Thanks [kungknut]! That's exactly what I was after!

Added that to my docker compose file, edited the version.php file in the nextcloud data directory to reflect version 17, then did a docker-compose pull and docker-compose up -d.

This got me to a working version 17 install, then I ran occ upgrade, fixed a few upgrade issues, put it back online and viola, I had a fully functioning version 17 instance.

Then I went back to docker compose, changed the image reference to the generic image and did another pull and up -d.

I am now the proud owner of a fully updated nextcloud 18 server! Thanks!
